dragonfly_doe2/doe/floor_space.py

```diff
@@ -0,0 +1,149 @@
+from dragonfly.room2d import Room2D
+from dragonfly.story import Story
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass
+class Wall:
+    """*.inp Wall object.
+    Example:
+        .. code-block::f#
+            "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+               CONSTRUCTION    = "EWall Construction"
+               LOCATION        = SPACE-V1
+               ..
+    """
+    name: str = None
+    location_index: int = None
+    wall_con_name: str = None
```

Comment:
My understanding is that these inputs are required. We should remove the default None value.

```suggestion
    name: str
    location_index: int
    wall_con_name: str
```

Reply:
when there is not a ```None```:
I get a __init__() missing args error, for instance on the Floor() class:
```python
>   flr_spc = [Floor().from_story(story) for story in df_model.stories]
E   TypeError: __init__() missing 4 required positional arguments: 'name', 'floor_z', 'floor_height', and 'spaces'
```
Would some generic: ```name: str = 'room_name'```, ```floor_height: float = 3.0``` be the move here or is this indicative of further human error?

Reply:
for instance:
```python
    name: str = ''
    floor_z: float = 0.0
    floor_height: float = 3.0
    spaces: List[Space] = None
 ```
 Works 4.0

Reply:
This happens because you are initiating the classmethod wrong. I added a comment below. Your code should be like this. No `()` after Floor.

```Python
flr_spc = [Floor.from_story(story) for story in df_model.stories]
```

Then you can remove the default values.

Reply:
OHHHHHHHH!!!